### DISCUSSION TITLE
Documentation for modifying appearance of cmdline_popup dropdown

### DISCUSSION THREAD

**Discussion by tj-smith47** [folke/noice.nvim#874] [Q&A]
Hello,

Maybe I'm posting in the wrong place, but there is a web of plugins and dependencies involved in what I'm trying to do. I know it also involves `telescope.nvim` and maybe `nvim-cmp`, but ultimately it tracks back to the `cmdline_popup` configuration and associated results, so this project is the most sensible place to ask for help.

I don't like how the dropdown / popupmenu / results (there are so many similar terms in the space, I am not even sure what term or setting I'm really looking for) resizes as I type, it's distracting. I'm trying to figure out how to get the results from the `cmdline_popup` to appear in a connected popup like the Code Actions do in the top photo from `telescope-ui-select.nvim`. The results could even be detached like they are in the bottom photo - as long as they're the same width as the `cmdline_popup` and appear underneath the input box:
![Screenshot 2024-06-23 at 16 27 59](https://github.com/folke/noice.nvim/assets/114961514/1196955f-ee74-4ef7-8671-c26cbc900402)
![image](https://github.com/folke/noice.nvim/assets/114961514/f30ed81f-9dc3-4e21-8aa5-35e1c6bce36f)
> I've seen other issues where you replied suggesting this is not possible, but I see nvim-cmp suggestions in the dropdown. I don't understand how there could be so many images like this out there if it weren't possible. I'm hoping it was just an issue at that time, and that now it is indeed possible.

 I've tried using `telescope-cmdline.nvim`, but it reverses the orientation and overrides many of the default behaviors I like about noice. I've spent days trying to figure out how to do this and looked at countless nvim configs without success, and am now caving writing this issue.  It seems like this behavior is driven by `nvim-cmp` maybe, but that the specific layout is some kind of telescope option, which just needs to be applied somewhere very, very specific. 

I would immensely appreciate any guidance you can give in helping me understand where I can control this config. If there's anything you need from me to assist, please let me know and I will be happy to provide it. Thanks for all you have contributed to the nvim community!

**Comment by tj-smith47** [comment]
To update - I was able to figure out how to get the width of the cmp popupmenu to stay a constant width following the bottom post in [this thread](https://github.com/hrsh7th/nvim-cmp/discussions/609).
<img width="1330" alt="Screenshot 2024-06-24 at 08 08 15" src="https://github.com/folke/noice.nvim/assets/114961514/4b98444e-a949-4fc9-83f9-d1ec922e834f">

Actual config change in nvim-cmp:
```lua
      opts = {
        format = function(entry, item)
          if vim.tbl_contains({ "path" }, entry.source.name) then
            local icon, hl_group = require("nvim-web-devicons").get_icon(entry:get_completion_item().label)
            if icon then
              item.kind = icon
              item.kind_hl_group = hl_group
              return item
            end
          end
          if vim.tbl_contains({ "cmdline" }, entry.source.name) then
            -- Get the completion entry text shown in the completion window.
            local content = item.abbr

            -- Set the fixed completion window width.
            local fixed_width = 119
            vim.o.pumwidth = fixed_width

            -- Get the width of the current window.
            local win_width = vim.api.nvim_win_get_width(0)

            -- Set the max content width based on either: 'fixed_width'
            -- or a percentage of the window width, in this case 20%.
            -- We subtract 10 from 'fixed_width' to leave room for 'kind' fields.
            local max_content_width = fixed_width and fixed_width - 10 or math.floor(win_width * 0.2)

            -- Truncate the completion entry text if it's longer than the
            -- max content width. We subtract 3 from the max content width
            -- to account for the "..." that will be appended to it.
            if #content > max_content_width then
              item.abbr = vim.fn.strcharpart(content, 0, max_content_width - 3) .. "..."
            else
              item.abbr = content .. (" "):rep(max_content_width - #content)
            end
          end
          return lspkind.cmp_format()(entry, item)
        end,
        -- Other configs ...
      }
```

With this, I'm sure I could get the top photo in my OP by just modifying the border properties of the cmp window for `cmdline` sources, but at this point I'm determined to get the detatched view with the offset. I'm almost there, but I'm still working on accounting for 2 things:
1. Setting the Y offset.
2. Adjusting the X offset when the `cmdline_popup` adjusts to hide text from `:lua`, `:help`, etc.

For the second one, I'm sure I can check what you're doing in the source code for this repo and copy that approach. I'll try to find it and attempt once I get the time. For the first bullet, maybe this one is easy. Just starting to really learn lua after having to ditch Packer for lazy when I moved to nvim 0.10.0. I will do my best to dig into this and share the changes I had to make once I get it. Any guidance / wisdom you can share to point me in the right direction you might have would be immensely appreciated! 🙏

**Reply by chris4prez** [reply]
I think I'm running down the same path as you. Were you able to sort this out and find a solution to achieve a cohesive auto completion where both the cmdline and auto completion from cmp maintain a similar width and look, as well as a offset gap between the two? Similar to how it functions without auto completion enabled?
